diff --git a/node_modules/@root/greenlock/greenlock.js b/node_modules/@root/greenlock/greenlock.js
index b346f38..2e0f59a 100644
--- a/node_modules/@root/greenlock/greenlock.js
+++ b/node_modules/@root/greenlock/greenlock.js
@@ -147,7 +147,8 @@ G.create = function(gconf) {
         return p;
     };
 
-    greenlock.notify = greenlock._notify = function(ev, params) {
+    greenlock.notify = greenlock._notify = async function(ev, params) {
+        if ('error' === ev) console.trace('notify error: ', params);
         var mng = greenlock.manager;
 
         if ('_' === String(ev)[0]) {
