diff --git a/node_modules/component-classes/index.js b/node_modules/component-classes/index.js
index eb9d292..a484984 100644
--- a/node_modules/component-classes/index.js
+++ b/node_modules/component-classes/index.js
@@ -2,11 +2,7 @@
  * Module dependencies.
  */
 
-try {
-  var index = require('indexof');
-} catch (err) {
-  var index = require('component-indexof');
-}
+var index = require('component-indexof');
 
 /**
  * Whitespace regexp.
