diff --git a/lib/components/ColorPicker.js b/lib/components/ColorPicker.js
index 26a96b5..537affe 100644
--- a/lib/components/ColorPicker.js
+++ b/lib/components/ColorPicker.js
@@ -121,6 +121,7 @@ var ColorPickerField = function ColorPickerField(_ref3) {
   }, restInput, restProps));
 };
 
+exports.ColorPickerControllable = ColorPicker;
 exports.ColorPickerField = ColorPickerField;
 ColorPickerField.propTypes = {
   input: _propTypes["default"].object,
