diff --git a/index.js b/index.js
index 39280a905941f391367c1aeca5ca3c5fd3749d52..9d13f9f99ea1b5460787b1a6818fa70628404410 100644
--- a/index.js
+++ b/index.js
@@ -84,7 +84,8 @@ var SwaggerUI = /*#__PURE__*/function (_React$Component) {
         filter: _includesInstanceProperty(_context2 = ["boolean", "string"]).call(_context2, _typeof(_filterInstanceProperty(this.props))) ? _filterInstanceProperty(this.props) : false,
         persistAuthorization: this.props.persistAuthorization,
         withCredentials: this.props.withCredentials,
-        oauth2RedirectUrl: this.props.oauth2RedirectUrl
+        oauth2RedirectUrl: this.props.oauth2RedirectUrl,
+        parameterMacro: this.props.parameterMacro
       });
       this.system = ui;
       this.SwaggerUIComponent = ui.getComponent("App", "root");
@@ -154,7 +155,8 @@ SwaggerUI.propTypes = {
   displayRequestDuration: PropTypes.bool,
   persistAuthorization: PropTypes.bool,
   withCredentials: PropTypes.bool,
-  oauth2RedirectUrl: PropTypes.string
+  oauth2RedirectUrl: PropTypes.string,
+  parameterMacro: PropTypes.func
 };
 SwaggerUI.defaultProps = {
   layout: "BaseLayout",